src/libYARP_sig/include/yarp/sig/Vector.h

```diff
@@ -277,7 +308,28 @@ class yarp::sig::VectorOf : public VectorBase
         bytes.allocateOnNeed(bytes.used()+sizeof(T),bytes.length()*2+sizeof(T));
         bytes.setUsed(bytes.used()+sizeof(T));
         _updatePointers();
-        first[len-1] = elem;
+        new (&((*this)[len-1])) T(elem); // non-allocating placement operator new
+    }
+
+    /**
+    * Move a new element in the vector: size is changed
```

Comment:
Very minor issue... Check the alignment of the `*` in this and in the following comment